### ISSUE TITLE
Automatically add `-- ` when pressing enter in a comment

### ISSUE THREAD

**Issue by Sir4ur0n** [haskell/vscode-haskell#279]
A minor but nice feature I would like to propose is to automatically add `-- ` on the new line when pressing "enter" in a comment/documentation.

Example (cursor position represented by `[]`) below.

# Current behavior
```haskell
-- | Somme doc[]
foo = "hello world"
```
Press enter
```haskell
-- | Somme doc
[]
foo = "hello world"
```

# Proposed behavior
```haskell
-- | Somme doc[]
foo = "hello world"
```
Press enter
```haskell
-- | Somme doc
-- []
foo = "hello world"
```

I think this default behavior makes much more sense for developers.

To be discussed: should we do the same when in a regular comment? I'd say "no" but I am not strongly opinionated on this one.

Cheers

**Comment by and-pete** [comment]
Would it be possible to have a setting in the VSCode Haskell extension to select which other VSCode extension is responsible for the Haskell syntax highlighting/etc. in the same way that there is for the Formatting Provider?

I've currently got a fork of `JustusAdam/language-haskell` with a range of domain-specific syntax highlighting choices (mainly around quasi-quotations of different types). But the choice to use that highlighting fork disables the VSCode Haskell server.

**Comment by alanz** [comment]
FYI, LSP does support a `textDocument/onTypeFormatting` request, it might be possible to use it for this.

**Comment by bubba** [comment]
See https://github.com/JustusAdam/language-haskell/pull/174

**Comment by jneira** [comment]
The issue is blocked on the pull request linked above

**Comment by dlight** [comment]
So https://github.com/haskell/vscode-haskell/pull/174 wasn't merged, and JustusAdam/language-haskell#186 (the followup PR) wasn't either, so.. the extension currently doesn't have this feature, right? (Or is there another PR I'm missing?)
